### PR TITLE
Comment Replies

### PR DESCRIPTION
**Pull request by grantyang**
- Add reply button and inputField to each comment
- Refactor CommentInput to handle new replies to song comments
- Refactor CommentList to display replies to song comments
- Add comments to clarify logic

### CODE REVIEW

File: client/src/components/CommentInput.jsx [hunk 1]
```diff
@@ -21,43 +22,66 @@ class CommentInput extends Component {
   }
 
   onInputSubmit(event) {
+    console.log(this.props);
```
Comment (Jshhhh) [inline client/src/components/CommentInput.jsx]: Can remove console.log

File: client/src/components/CommentInput.jsx [hunk 2]
```diff
@@ -21,43 +22,66 @@ class CommentInput extends Component {
...
           <input type="text" value={this.state.inputValue} onChange={this.onInputChange} placeholder="Write a comment" />
         </form>
+      );
+    } else if (this.state.commentType === 'reply') {
+      inputForm = (
+        <form onSubmit={this.onInputSubmit}>
+          <input type="text" value={this.state.inputValue} onChange={this.onInputChange} placeholder="Write a reply" />
+        </form>
+      );
+    }
```
Comment (Jshhhh) [inline client/src/components/CommentInput.jsx]: Possibly can use a ternary operator for placeholder if you're only expecting 'reply' and 'song' for state.commentType. Saves you a few lines of code, but this works fine. Up to you.

File: database/database.js [hunk 3]
```diff
@@ -32,9 +32,15 @@ const addSongComment = (songId, authorId, text, parentId, timestamp, callback) =
   connection.query(
     'INSERT INTO comments (song_id, author_id, comment, parent_id, timestamp) values (?, ?, ?, ?, ?)',
     [songId, authorId, text, parentId, timestamp],
-    (error, results) => {
+    (error) => {
       if (error) callback(error, null);
-      else callback(null, results);
+      connection.query(
```
Comment (Jshhhh) [inline database/database.js]: Doesn't this query still invoke even if error?

**Comment by grantyang** [comment]
Made requested changes. Please approve if you're happy with them.

**Pull request merged**